Show prominent update notification on the status bar when there is a new update available (Run ID: codestoryai_aide_issue_1255_a5b6dc56)

theskcd opened the pull request:
agent_instance: codestoryai_aide_issue_1255_a5b6dc56 Tries to fix: #1255

sartoshi-foot-dao commented:
/agent try again

theskcd commented:
I've pushed a new commit (864b07e) to this PR. Please review the changes.

sartoshi-foot-dao commented:
/agent continue

theskcd commented:
I've pushed a new commit (25f1129) to this PR. Please review the changes.

sartoshi-foot-dao commented:
/q what did you just do?

theskcd commented:
I enhanced VS Code's update status bar notifications with more detailed states and visual indicators. Specifically, I made these improvements:

1. Added distinct visual states for different update phases:
- "Ready to Update" state with red background and spinning icon
- "Update Downloaded" state with yellow background
- "Installing Update" state with spinning icon
- "Update Available" state with default styling

2. Improved tooltips with clearer instructions:
- Made tooltip text more descriptive by explicitly mentioning what action will happen (e.g., "Click to restart and install the update" instead of just "Click to install")
- Added specific tooltip for the installing state

3. Added proper error handling and state management:
- Added handling for the Updating state to show installation progress
- Disabled click action during installation by removing the command when updating
- Added cleanup for Uninitialized state
- Clear status bar entry when updates are complete, disabled, or uninitialized

4. Enhanced visual styling:
- Used proper theme colors (error/warning) for important states
- Added prominent styling using the 'kind' property for important states
- Used consistent icons (sync~spin for progress, cloud-download for updates)

These changes make the update process more informative and user-friendly by providing clearer visual feedback and instructions throughout the different stages of the update lifecycle.

ghostwriternr commented:
/agent I'm seeing some errors:
```
[watch-client    ] [16:15:26] Error: /home/ghost_000/github/codestory/aide/src/vs/workbench/contrib/update/browser/update.ts(22,10): Module '"../../../../platform/theme/common/themeService.js"' has no exported member 'ThemeColor'.
[watch-client    ] [16:15:26] Error: /home/ghost_000/github/codestory/aide/src/vs/workbench/contrib/update/browser/update.ts(181,4): Cannot find name 'IStatusbarService'.
[watch-client    ] [16:15:26] Error: /home/ghost_000/github/codestory/aide/src/vs/workbench/contrib/update/browser/update.ts(181,57): Cannot find name 'IStatusbarService'.
[watch-client    ] [16:15:26] Error: /home/ghost_000/github/codestory/aide/src/vs/workbench/contrib/update/browser/update.ts(255,5): Cannot find name 'StatusbarAlignment'
```

theskcd commented:
I've updated the imports in update.ts to fix the type errors. Specifically, I imported ThemeColor from the correct path ('../../../../base/common/themables.js') and added the missing IStatusbarService and StatusbarAlignment imports from the statusbar service. Could you please review these changes to ensure they properly resolve the type errors while maintaining the status bar update notifications functionality?

ghostwriternr commented:
/agent run npm run compile and verify your work
